src/components/SignUpForm/index.test.js

```diff
@@ -0,0 +1,409 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+import SignUpForm from "./index";
+
+describe("SignUpForm", () => {
+  it("renders correctly and matches the snapshot", () => {
+    const tree = renderer.create(<SignUpForm />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  const mockSubmit = jest.fn(
+    (firstName, lastName, email, password, confirmPassword, acceptTerms) => {
+      return Promise.resolve({
+        firstName,
+        lastName,
+        email,
+        password,
+        confirmPassword,
+        acceptTerms,
+      });
+    }
+  );
+
+  beforeEach(() => {
+    render(<SignUpForm submit={mockSubmit} />);
+  });
+
+  it("should display required error when value is empty", async () => {
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(5);
+    expect(mockSubmit).not.toBeCalled();
+  });
+
+  it("should display matching error when email is invalid", async () => {
+    fireEvent.input(screen.getByRole("textbox", { name: /firstName/i }), {
+      target: {
+        value: "name",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /lastName/i }), {
+      target: {
+        value: "surname",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/^password/i), {
+      target: {
+        value: "password",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/repeat password/i), {
+      target: {
+        value: "password",
+      },
+    });
+
```

Comment:
these tests need more comments. I'd say one for each user action you're mimicking. also this usage of getbyrole checkbox is bad testing practice because you are assuming there is only one checkbox. if your teammate adds another one, such as a checkbox to sign up for a newsletter, these tests will break. 

this applies for all getbyrole or similar calls where you are currently relying on the fact that there is only one of them (if any)

Reply:
you are totally right, I fixed that, thank you very much